products/jbrowse-cli/src/commands/text-index.ts

```diff
@@ -35,61 +35,76 @@ export default class TextIndex extends JBrowseCommand {
     }),
+
+    attributes: flags.string({
+      description: 'Comma separated list of attributes to index',
+      default: 'Name,ID',
+    }),
     assemblies: flags.string({
       char: 'a',
       description:
         'Specify the assembl(ies) to create an index for. If unspecified, creates an index for each assembly in the config',
     }),
+    force: flags.boolean({
+      default: false,
+      description: 'Overwrite previously existing indexes',
+    }),
+    quiet: flags.boolean({
+      char: 'q',
+      default: false,
+      description: 'Hide the progress bars',
+    }),
   }
 
   // Called when running the terminal command. Parses the given flags and
   // tracks associated. Gets their information and sends it to the appropriate
   // file parser to be indexed
   async run() {
-    const defaultAttributes = ['Name', 'description', 'Note', 'ID']
-    const { flags } = this.parse(TextIndex)
-    const outdir = flags.target || flags.out || '.'
-    const isDir = (await fs.promises.lstat(outdir)).isDirectory()
-    const target = isDir ? `${outdir}/config.json` : outdir
-
-    const configDirectory = path.dirname(target)
-    const config: Config = JSON.parse(fs.readFileSync(target, 'utf8'))
+    const {
+      flags: { out, target, tracks, assemblies, attributes, quiet, force },
+    } = this.parse(TextIndex)
+    const outFlag = target || out || '.'
+    const confFile = fs.lstatSync(outFlag).isDirectory()
+      ? path.join(outFlag, 'config.json')
+      : outFlag
+    const dir = path.dirname(confFile)
+
+    const config: Config = JSON.parse(fs.readFileSync(confFile, 'utf8'))
     const assembliesToIndex =
-      flags.assemblies?.split(',') || config.assemblies?.map(a => a.name) || []
+      assemblies?.split(',') || config.assemblies?.map(a => a.name) || []
     const adapters = config.aggregateTextSearchAdapters || []
 
-    const hash: string = crypto
-      .createHash('md5')
-      .update(
-        assembliesToIndex.toString() + flags.tracks?.split(','.toString()),
-      )
-      .digest('base64')
+    const trixDir = path.join(dir, 'trix')
+    if (!fs.existsSync(trixDir)) {
+      fs.mkdirSync(trixDir)
+    }
 
     for (const asm of assembliesToIndex) {
-      const config = await this.getConfig(target, asm, flags.tracks?.split(','))
+      const config = await this.getConfig(confFile, asm, tracks?.split(','))
 
       this.log('Indexing assembly ' + asm + '...')
 
       if (config.length) {
-        await this.indexDriver(config, defaultAttributes, configDirectory, asm)
+        const id = asm + '-index'
+        const adapterAlreadyFound = adapters.find(
+          x => x.textSearchAdapterId === id,
+        )
+        if (adapterAlreadyFound && !force) {
+          this.log(
+            `Note: ${asm} has already been indexed with this configuration, use --force to overwrite this assembly. Skipping for now`,
+          )
+        }
+
+        await this.indexDriver(config, attributes.split(','), dir, asm, quiet)
```

Comment:
need to have the attributes come from the track configs

Reply:
I think it's actually better to make it not use track config, and use CLI flags.

The track configs goes back to JBrowse 1 style "--nameAttributes in flatfile-to-json loading time", which I think is more confusing than letting generate-names do it.

Reply:
And then, we'd have to add --nameAttributes essentially to the jbrowse 2 CLI.

Reply:
If it is really needed to make a specific track have different attributes, we could either

0) not use --attributes at all, not really my preferred option
a) make --attributes a default and per-track setting is an override as @TiaSFlores mentions
b) make the user index that track separately as a per track index
c) add --incremental indexing to make them add the track as an extra step to their aggregate index

Reply:
yeah it would be reasonable for `--attributes` to override the settings from the track configs. Just talking now with @TiaSFlores and @teresam856 about this. We just need to make sure that the attributes that were actually used is recorded in the meta.json so we can reason about them in the front end

Reply:
why does the front end need to know what attribute was indexed necessarily?